Sources/eips/eip1024/EthEncryptedData+EIP1024.swift

```diff
@@ -51,12 +95,72 @@ extension EthEncryptedData {
         
         return message
     }
+    
+    
+    /// Decrypt the message using the sender's private key
+    /// - Parameters:
+    ///   - senderPrivateKey: sender private key as PrivateKeyEth1
+    ///   - recipientPublicKey: string of recipient's public key
+    /// - Returns: cleartext message
+    public func decrypt(senderPrivateKey: PrivateKeyEth1, recipientPublicKey: String) throws -> String {
+        guard let privateKey = senderPrivateKey.string() else { throw EthCryptoError.keyError }
+        return try decrypt(senderPrivateKey: privateKey, recipientPublicKey: recipientPublicKey)
+    }
+    
+    /// Decrypt the message using the sender's private key
+    /// - Parameters:
+    ///   - senderPrivateKey: sender private key as string
+    ///   - recipientPublicKey: string of recipient's public key
+    /// - Returns: cleartext message
+    public func decrypt(senderPrivateKey: String, recipientPublicKey: String) throws -> String {
+        let data = Data(hex: senderPrivateKey)
+                
+        let secretKey = try TweetNacl.keyPair(fromSecretKey: data).secretKey
+        
+        guard let nonce = Data(base64Encoded: self.nonce),
+              let cipherText = Data(base64Encoded: self.ciphertext),
+              let recipientPublicKeyData = Data(base64Encoded: recipientPublicKey) else {
+          throw EthCryptoError.decryptionFailed
+        }
+        
+        let decrypted = try TweetNacl.open(
+            message: cipherText,
+            nonce: nonce,
+            publicKey: recipientPublicKeyData,
+            secretKey: secretKey
+        )
+        
+        guard let message = String(data: decrypted, encoding: .utf8) else {
+            throw EthCryptoError.decryptionFailed
+        }
+        
+        return message
+    }
 }
 
 extension PrivateKeyEth1 {
-  public func eth_publicKey() throws -> String {
```

Comment:
@ronaldmannak can you please keep this method to not break back compatibility, but mark it as 'deprecated'?